Sources/ScipioKit/ProjectGenerator/ProjectGenerator.swift

```diff
@@ -0,0 +1,505 @@
+            """)
+        }
+
+        let resourcePhase = addObject(
+            try makeResourcePhase(for: target, targetGroup: targetGroup)
+        )
+
+        // TODO generate resource accessor to support Bundle.module
+        // https://github.com/apple/swift-package-manager/blob/29a16bc2dc0ef72b7044c1dc6236236e3d0120e0/Sources/Build/BuildPlan.swift#L806-L852
+
+        return PBXNativeTarget(name: target.c99name,
+                               buildConfigurationList: buildConfigurationList,
+                               buildPhases: [compilePhase, resourcePhase],
+                               product: productRef,
+                               productType: productType)
+    }
+
+    private func makeResourcePhase(for target: ResolvedTarget, targetGroup: PBXGroup) throws -> PBXResourcesBuildPhase {
+        guard let sourceRoot else {
+            throw Error.invalidPackage(packageName: package.name)
+        }
+
+        let resourcesReferences: [PBXFileReference] = try target.underlyingTarget.resources.reduce([]) { lists, resource in
+            let resourcesGroup = try targetGroup.addGroup(named: "Resources").first!
+
+            switch resource.rule {
+            case .process(let localization):
+                guard localization == nil else {
+                    throw Error.localizationNotSupported(targetName: target.name)
+                }
+            case .copy:
+                break
+            }
+
+            let files = try walk(resource.path)
+            return try files.map { file in
+                return try resourcesGroup.addFile(at: file.toPath(), sourceRoot: sourceRoot.toPath())
+            } + lists
+        }
+
+        let buildFiles = resourcesReferences.map { PBXBuildFile(file: $0) }
+
+        return PBXResourcesBuildPhase(files: buildFiles)
+    }
+
+    private struct ClangTargetSettingResult {
+        var moduleMapPath: AbsolutePath?
+    }
+
+    private func applyClangTargetSpecificSettings(for clangTarget: ClangTarget, xcodeTarget: PBXNativeTarget) throws -> ClangTargetSettingResult {
+        guard let mainGroup = pbxProj.rootObject?.mainGroup else {
+            throw Error.unknownError
+        }
+        let targetGroup = createOrGetGroup(named: clangTarget.name, in: mainGroup, path: clangTarget.sources.root)
+
+        let includeDir = clangTarget.includeDir
+        let headerFiles = try walk(includeDir).filter { $0.extension == "h" }
+        var headerFileRefs: [PBXFileReference] = []
+        for header in headerFiles {
+            let headerFileGroup = try self.group(
+                for: header.parentDirectory,
+                parentGroup: targetGroup,
+                sourceRoot: clangTarget.sources.root
+            )
+            let fileRef = try headerFileGroup.addFile(at: header.toPath(),
+                                                      sourceRoot: clangTarget.path.toPath())
+            headerFileRefs.append(fileRef)
+        }
```

Comment:
`headerFileRefs` can become `let`
```suggestion
        let headerFileRefs: [PBXFileReference] = try headerFiles.map { header in
            let headerFileGroup = try self.group(
                for: header.parentDirectory,
                parentGroup: targetGroup,
                sourceRoot: clangTarget.sources.root
            )
            return try headerFileGroup.addFile(at: header.toPath(), sourceRoot: clangTarget.path.toPath())
        }
```

Reply:
I don't like to have side-effects inside `map` a little....